### PR TITLE
fix(SPV-769): check sql.select result length

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/block-headers-service/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/block-headers-service/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: service/header_service.go [hunk 1]
```diff
@@ -66,12 +66,12 @@ func (hs *HeaderService) IsCurrent() bool {
 	// Not current if the latest main (best) chain height is before the
 	// latest known good checkpoint (when checkpoints are enabled).
 	checkpoints := hs.checkpoints
-	checkpoint := &checkpoints[len(checkpoints)-1]
+	checkpoint := checkpoints[len(checkpoints)-1]
```
Comment (chris-4chain) [inline service/header_service.go:69]: What if the `checkpoints` is empty?

**Review by chris-4chain — APPROVED** [review]

File: service/header_service.go (hunk 1 above)
Comment (arkadiuszos4chain) [reply service/header_service.go]: I have no idea what are you talking about :P Change reverted - out of the scope

**Review by Nazarii-4chain — APPROVED** [review]

**Review by wregulski — APPROVED** [review]

**Review by kuba-4chain — COMMENTED** [review]
In the method `GetAllTips()` in this file, we're also using `db.Select()`. If we're deciding on this approach of returning an error when we retrieve no records, then maybe we should add that in this `GetAllTips()` method as well?

**Comment by jakubmkowalski** [comment]
testing

**Comment by dorzepowski** [comment]
> In the method `GetAllTips()` in this file, we're also using `db.Select()`. If we're deciding on this approach of returning an error when we retrieve no records, then maybe we should add that in this `GetAllTips()` method as well?

Those changed methods where we're returning errors are semantically expected to return something, so when we don't have this something it is expected to get an error. 
While in case of GetAll**** it is expected to get all and if there isn't anything then get empty list. 
(It is a common design for such methods)

**Pull request merged**